In [1]:
import pandas as pd
from datetime import datetime

In [2]:
df= pd.read_csv('BKB_WaterQualityData_2020084.csv')

In [3]:
df

Site_Id Unit_Id   Read_Date  Salinity (ppt)  Dissolved Oxygen (mg/L)  \
0        Bay     NaN    1/3/1994             1.3                     11.7   
1        Bay     NaN   1/31/1994             1.5                     12.0   
2        Bay     NaN    2/7/1994             1.0                     10.5   
3        Bay     NaN   2/23/1994             1.0                     10.1   
4        Bay     NaN   2/28/1994             1.0                     12.6   
...      ...     ...         ...             ...                      ...   
2366     Bay     NaN  10/11/2018             1.9                      5.0   
2367     Bay     NaN  10/24/2018             0.0                      9.0   
2368     Bay     NaN  10/28/2018             0.9                      2.9   
2369     Bay     NaN   11/7/2018             1.7                      NaN   
2370     Bay     NaN  12/11/2018             0.1                      NaN   

      pH (standard units)  Secchi Depth (m)  Water Depth (m)  Water Temp (?C)  \
0                     7.3              0.40             0.40              5.9   
1                     7.4              0.20             0.35              3.0   
2                     7.2              0.25             0.60              5.9   
3                     7.4              0.35             0.50             10.0   
4                     7.2              0.20             0.40              1.6   
...                   ...               ...              ...              ...   
2366                  7.0              4.00             1.20             25.0   
2367                  7.0              0.30             0.60             18.0   
2368                  7.0              0.40             0.90             13.0   
2369                  7.0              0.45             0.90             20.0   
2370                  7.0              0.10             0.10             10.0   

      Air Temp-Celsius  Air Temp (?F) Time (24:00) Field_Tech DateVerified  \
0                  8.0          46.40        11:00        NaN          NaN   
1                  2.6          36.68        11:30        NaN          NaN   
2                  7.6          45.68         9:45        NaN          NaN   
3                  2.7          36.86          NaN        NaN          NaN   
4                  0.0          32.00        10:30        NaN          NaN   
...                ...            ...          ...        ...          ...   
2366               NaN          78.00        09:30    Sue Poe   11/13/2019   
2367               NaN          58.00        09:30    Sue Poe   11/13/2019   
2368               NaN          49.00        09:20    Sue Poe   11/13/2019   
2369               NaN          65.00        09:45    Sue Poe   11/13/2019   
2370               NaN          42.00        09:40    Sue Poe   11/13/2019   

          WhoVerified  AirTemp (C)  Year  
0                 NaN     8.000000  1994  
1                 NaN     2.600000  1994  
2                 NaN     7.600000  1994  
3                 NaN     2.700000  1994  
4                 NaN     0.000000  1994  
...               ...          ...   ...  
2366  Christine Folks    25.555556  2018  
2367  Christine Folks    14.444444  2018  
2368  Christine Folks     9.444444  2018  
2369  Christine Folks    18.333333  2018  
2370  Christine Folks     5.555556  2018  

[2371 rows x 17 columns]

In [4]:
# t= df[(df['Read_Date'] >= '1/1/1998') & (df['Read_Date'] <= '12/31/2018')]
t=df.query("Year >=1998 and Year <= 2018")

In [5]:
t

Site_Id Unit_Id   Read_Date  Salinity (ppt)  Dissolved Oxygen (mg/L)  \
330      Bay     NaN    1/4/1998             1.5                     12.3   
331      Bay     NaN   1/25/1998             1.2                     12.5   
332      Bay     NaN   2/22/1998             1.1                     10.7   
333      Bay     NaN   3/22/1998             1.0                     11.1   
334      Bay     NaN   4/19/1998             1.0                     10.8   
...      ...     ...         ...             ...                      ...   
2366     Bay     NaN  10/11/2018             1.9                      5.0   
2367     Bay     NaN  10/24/2018             0.0                      9.0   
2368     Bay     NaN  10/28/2018             0.9                      2.9   
2369     Bay     NaN   11/7/2018             1.7                      NaN   
2370     Bay     NaN  12/11/2018             0.1                      NaN   

      pH (standard units)  Secchi Depth (m)  Water Depth (m)  Water Temp (?C)  \
330                   6.4              0.25             0.50              8.9   
331                   6.5              0.30             0.60              8.1   
332                   6.3              0.25             0.75             10.5   
333                   6.4              0.25             0.71             11.0   
334                   6.9              0.30             0.78             17.0   
...                   ...               ...              ...              ...   
2366                  7.0              4.00             1.20             25.0   
2367                  7.0              0.30             0.60             18.0   
2368                  7.0              0.40             0.90             13.0   
2369                  7.0              0.45             0.90             20.0   
2370                  7.0              0.10             0.10             10.0   

      Air Temp-Celsius  Air Temp (?F) Time (24:00) Field_Tech DateVerified  \
330                NaN          57.20         4:48    Strader          NaN   
331                NaN          45.86        20:38    Strader          NaN   
332                NaN            NaN        12:00    Strader          NaN   
333                NaN          48.74        17:45    Strader          NaN   
334                NaN            NaN         0:00    Strader          NaN   
...                ...            ...          ...        ...          ...   
2366               NaN          78.00        09:30    Sue Poe   11/13/2019   
2367               NaN          58.00        09:30    Sue Poe   11/13/2019   
2368               NaN          49.00        09:20    Sue Poe   11/13/2019   
2369               NaN          65.00        09:45    Sue Poe   11/13/2019   
2370               NaN          42.00        09:40    Sue Poe   11/13/2019   

          WhoVerified  AirTemp (C)  Year  
330               NaN    14.000000  1998  
331               NaN     7.700000  1998  
332               NaN   -17.777778  1998  
333               NaN     9.300000  1998  
334               NaN   -17.777778  1998  
...               ...          ...   ...  
2366  Christine Folks    25.555556  2018  
2367  Christine Folks    14.444444  2018  
2368  Christine Folks     9.444444  2018  
2369  Christine Folks    18.333333  2018  
2370  Christine Folks     5.555556  2018  

[1971 rows x 17 columns]

In [6]:
t.to_csv('back_bay_1998-2018_clean.csv')

In [7]:
print(len(df.columns))

17


In [8]:
print(t.columns)

Index(['Site_Id', 'Unit_Id', 'Read_Date', 'Salinity (ppt)',
       'Dissolved Oxygen (mg/L)', 'pH (standard units)', 'Secchi Depth (m)',
       'Water Depth (m)', 'Water Temp (?C)', 'Air Temp-Celsius',
       'Air Temp (?F)', 'Time (24:00)', 'Field_Tech', 'DateVerified',
       'WhoVerified', 'AirTemp (C)', 'Year'],
      dtype='object')


In [9]:
t_mean = df["Dissolved Oxygen (mg/L)"].mean()

In [10]:
t_mean

6.646263157894744

In [11]:
column = t["AirTemp (C)"]
max_value = column.max()
max_value

33.0

In [12]:
t['AirTemp (C)'].min()

-17.77777778

In [13]:
t.iloc[:, [2]].min()

Read_Date    1/10/2002
dtype: object

In [14]:
t["Year"].value_counts()

2009    120
2004    119
2006    118
2008    117
2001    115
2007    114
2010    109
2000    109
2002    108
2005    107
2003     98
2011     97
2013     91
2012     88
2015     88
2014     87
2017     85
2018     82
2016     59
1999     45
1998     15
Name: Year, dtype: int64

In [29]:
salinity_BySite =t[['Salinity (ppt)','Site_Id']]

In [32]:
print("Bay", salinity_BySite.query("Site_Id == 'Bay'").mean())
print("A", salinity_BySite.query("Site_Id == 'A'").mean())
print("B", salinity_BySite.query("Site_Id == 'B'").mean())
print("C", salinity_BySite.query("Site_Id == 'C'").mean())
print("D", salinity_BySite.query("Site_Id == 'D'").mean())
print("d", salinity_BySite.query("Site_Id == 'd'").mean())

Bay Salinity (ppt)    0.650556
dtype: float64
A Salinity (ppt)    0.334135
dtype: float64
B Salinity (ppt)    0.444838
dtype: float64
C Salinity (ppt)    0.596397
dtype: float64
D Salinity (ppt)    0.103483
dtype: float64
d Salinity (ppt)    0.0
dtype: float64


2. When looking at the mean ppt, which location would you infer is furthest from the influence of ocean water
inflows?

### Answer: 
As per the above data, area 'D' with mean ppt 0.103483 is the location which is furthest from the influence of ocean water. 

3. Dig a little deeper into #2, and write why there may be some uncertainty in your answer? (hint: certainty
is improved by consistency in data)


### Answer:

The data we have is unconsistence. on August 11, salinity ppt for Site D was .02. Again after 7 day later ppt level is 3 for the same site. This is a massive changes. It is hard to make decission based on the given data 

4. Use the data to determine the correlation between Salinity (ppt) and pH (standard units). Use the
DataFrame.corr(). You just need to report the correlation value

In [33]:
column_1 = t["Salinity (ppt)"]
column_2 = t["pH (standard units)"]
correlation = column_1.corr(column_2)
print(correlation)
    

0.26930634452336155
